notebooks/Emukit_task_seir_model/gillespie_analysis.py

```diff
@@ -0,0 +1,158 @@
+        """
+        :param alpha: the ratio of infection rate and recovery rate
+        :return: the probability density at alpha
+        """
+        return self.gamma.pdf(alpha)
+
+
+class UniformPrior(AlphaPrior):
+    """ Uniform prior on the infection/recovery rate parameter """
+    def __init__(self, alpha_min: float, alpha_max: float):
+        """
+        :param alpha_min: left interval bound
+        :param alpha_max: right interval bound
+        """
+        self.alpha_min = alpha_min
+        self.alpha_max = alpha_max
+        super(UniformPrior, self).__init__(name='uniform prior')
+
+    def evaluate(self, alpha: float) -> float:
+        """
+        :param alpha: the ratio of infection rate and recovery rate
+        :return: the probability density at alpha
+        """
+        return 1./(self.alpha_max - self.alpha_min)
+
+
+# statistics of gillespie runs
+class MeanMaxInfectionGillespie:
+    """
+    Statistics for the time occurrence and height of the infection peak of the gillespie simulation.
+    """
+    def __init__(self, gillespie_model: SIRGillespie, num_gil: int, time_end: float, alpha_prior: AlphaPrior):
+        """
+        :param gillespie_model: a SIRGillespie model
+        :param alpha_prior: the prior over alpha
+        :param num_gil: number of Gillespie samples to average over
+        :param time_end: end time of simulation
+        """
+        self.gillespie = gillespie_model
+        self.alpha_prior = alpha_prior
+        self.num_gil = num_gil
+        self.time_end = time_end
+
+    def evaluate_bare(self, alpha: float) -> Tuple[float, float]:
+        """
+        :param alpha: the ratio of infection rate and recovery rate
+        :return: Estimated (mean over num_gil simulations runs) time and height of the infection peak for given alpha.
+        """
+        self.gillespie.model.update_alpha(alpha)
+        peak_time, peak_height = self.gillespie.run_simulation_height_and_time_of_peak(self.num_gil, self.time_end)
+        return peak_time, peak_height
+
+    def evaluate(self, alpha: float) -> Tuple[float, float]:
+        """
+        :param alpha: the ratio of infection rate and recovery rate
+        :return: Estimated (mean over num_gil simulations runs) time and height of the infection peak for given alpha
+        scaled with the prior over alpha.
+        """
+        peak_time, peak_height = self.evaluate_bare(alpha)
+        alpha_weight = self.alpha_prior.evaluate(alpha)
+        return peak_time * alpha_weight, peak_height * alpha_weight
+
+    def evaluate_time_bare(self, alpha: float) -> float:
+        return self.evaluate_bare(alpha)[0]
+
+    def evaluate_height_bare(self, alpha: float) -> float:
+        return self.evaluate_bare(alpha)[1]
+
+    def evaluate_time(self, alpha: float) -> float:
+        return self.evaluate(alpha)[0]
+
+    def evaluate_height(self, alpha: float) -> float:
+        return self.evaluate(alpha)[1]
+
+
+# user functions
```

Comment:
I put everything into `gillespie_analysis.py` which is not directly related to the simulation but rather to the task such as defining priors and user functions. I could do separate files such as `seir_user_functions.py` and `alpha_priors.py` etc but I did not like that either since somewhat I only need the priors in combination with these specific user functions and so I put it all in one py-file.

The comments group into these categories.